src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -450,6 +450,30 @@ public bool Add(T item)
         }
 
         /// <summary>
+        /// Searches the set for a given value and returns the equal value it finds, if any.
+        /// </summary>
+        /// <param name="equalValue">The value to search for.</param>
+        /// <param name="actualValue">The value from the set that the search found, or the original value if the search yielded no match.</param>
+        /// <returns>A value indicating whether the search was successful.</returns>
+        /// <remarks>
+        /// This can be useful when you want to reuse a previously stored reference instead of 
+        /// a newly constructed one (so that more sharing of references can occur) or to look up
+        /// a value that has more complete data than the value you currently have, although their
+        /// comparer functions indicate they are equal.
+        /// </remarks>
+        public bool TryGetValue(T equalValue, out T actualValue)
+        {
+            int i = _buckets != null ? InternalIndexOf(equalValue) : -1;
+            if (i >= 0)
```

Comment:
Nit: it would be better to do:
```C#
if (_buckets != null)
{
    int i = InternalIndexOf(equalValue);
    if (i >= 0)
    {
        actualValue = _slots[i].value;
        return true;
    }
}
actualValue = default(T);
return false;
```
While the version you have is more concise, it introduces an unnecessary branch in the case of _buckets == null.